src/systems/filecoin_markets/deal/deal.id

```diff
@@ -34,6 +33,8 @@ type StorageDealProposal struct {
     ClientCollateralPerEpoch      actor.TokenAmount  // potentially collapse into one with ProviderDealCollateral
 
     ClientBalanceRequirement()    actor.TokenAmount  // (ClientCollateralPerEpoch + StoragePricePerEpoch) * (EndEpoch - StartEpoch)
+    TotalStorageFee()             actor.TokenAmount  // StoragePricePerEpoch * (EndEpoch - StartEpoch)
+    TotalClientCollateral()       actor.TokenAmount  // ClientCollateralPerEpoch * Duration
     ProviderBalanceRequirement()  actor.TokenAmount  // ProviderCollateralPerEpoch * (EndEpoch - StartEpoch)
     Duration()                    block.ChainEpoch
```

Comment:
Should we actually store duration? It seems misguided given start and end

Reply:
This is a helper function, we can call it FullDuration to be clear and precise.

Reply:
Just added the `@(cached)` call so we know it's just computed from other vlaies